lib/FileModel.php

```diff
@@ -43,51 +39,54 @@ public function analyze($currentDirectory) {
 			if ($item instanceof Folder) {
 				continue;
 			}
-			$this->pushToBiggestFilesIfNecessary($item);
+			$this->pushToBiggestFiles($item);
 			$this->analyzeMimeType($item);
+			usort($this->biggestFiles, function($a, $b) {
+			    return $a[1] < $b[1];
+            });
 		}
-		$this->convertBiggestFilesNodeToArray();
+
 	}
 	/**
+     * @var array $biggestFilesSorted
 	 * @return array
 	 */
 	public function getAnalysisReport() {
-		return [
-			'biggest_files' => $this->biggestFiles,
+        $this->analyze($this->userFolder);
+        $biggestFilesSorted = [];
+        if (sizeof($this->biggestFiles)<10) {
+            for ($i = 0; $i<sizeof($this->biggestFiles) ; $i++) {

```

Comment:
sizeof ile array count u bakmak saglikli degil, count fonksiyonu var bunun icin. Bir de burda iki ayri for yerine tek for ile halledebilirsin. for un limitini belirlerken biggestFiles in size'i ile default degeri kiyasla, kucuk olani kullan.